src/plone/api/content.py

```diff
@@ -67,7 +70,18 @@ def create(
     :Example: :ref:`content-create-example`
     """
     # Create a temporary id if the id is not given
-    content_id = not safe_id and id or str(random.randint(0, 99999999))
+    if not safe_id and id:
```

Comment:
Is there a reason to use an int instead of https://docs.python.org/3/library/uuid.html#uuid.uuid4 from the Python standard library? I don't know the underlying data type.

See the introduction from https://docs.python.org/3/library/uuid.html

> If all you want is a unique ID, you should probably call [uuid1()](https://docs.python.org/3/library/uuid.html#uuid.uuid1) or [uuid4()](https://docs.python.org/3/library/uuid.html#uuid.uuid4). Note that [uuid1()](https://docs.python.org/3/library/uuid.html#uuid.uuid1) may compromise privacy since it creates a UUID containing the computer’s network address. [uuid4()](https://docs.python.org/3/library/uuid.html#uuid.uuid4) creates a random UUID.

Reply:
I kept the underlying logic same...

Reply:
Since we are touching this, I would also be in favour of `str(uuid.uuid4())`.

Reply:
I would not go for a 32 characters id.

If you really need to use a UUID, I propose:
```
>>> uuid.uuid4().hex[:8]
'bba42d98'
```

Reply:
> >>> uuid.uuid4().hex[:8]
> 'bba42d98'

Is that guaranteed to be unique? The goal would be to remove the while loop to ensure uniqueness.

Reply:
> > > > > uuid.uuid4().hex[:8]
> > > > > 'bba42d98'
> 
> Is that guaranteed to be unique? The goal would be to remove the while loop to ensure uniqueness.

No, it can never be guaranteed to be unique, not even if you use the 32 character long one.
But we have 16^8 values to pick from, which is something like 2 ^ 32 which is more that 10 ^ 9.

Reply:
> > > > > > uuid.uuid4().hex[:8]
> > > > > > 'bba42d98'
> > 
> > Is that guaranteed to be unique? The goal would be to remove the while loop to ensure uniqueness.
> 
> No, it can never be guaranteed to be unique, not even if you use the 32 character long one.
> But we have 16^8 values to pick from, which is something like 2 ^ 32 which is more that 10 ^ 9.

Which in my opinion will be enough!

Reply:
Keep in mind that you're evaluating against millions of objects, not just one. See the [birthday paradox](https://en.wikipedia.org/wiki/Birthday_problem).

I also don't like playing odds, and prefer certainty. [uuid4()](https://docs.python.org/3/library/uuid.html#uuid.uuid4) is unique.

What's unmanageable about it? You click or tap the URL, and you're done.

Reply:
Remember that the id is temporary.  A few lines further on, we use the title to generate a nicer id.  So in my opinion the temporary id can be as ugly as we want.  So let's use the full uuid4.  And sure, retry 100 times if needed.

Reply:
done

Reply:
> Remember that the id is temporary. A few lines further on, we use the title to generate a nicer id. So in my opinion the temporary id can be as ugly as we want. So let's use the full uuid4. And sure, retry 100 times if needed.

I am pretty sure that I saw some content type (probably some that did not require a title), for which the generated id was kept.
BTW, I do not want to make this an argument. We anyway need to change the logic to [avoid moving the object after its creation](https://github.com/plone/plone.api/issues/439).
But that is happening in another PR.